### PR TITLE
Add an optional userID to Entity constructors.

### PR DESCRIPTION
**Pull request by shans**
This will be used as the last component of the ID if specified. Note that you are responsible for providing unique IDs if this facility is used.

UserIDs are namespaced separately from automatic IDs, both to prevent accidental collision and to provide for extraction of the ID when read from a handle.

### CODE REVIEW

File: runtime/viewlet.js [hunk 1]
```diff
@@ -61,12 +61,18 @@ class Viewlet {
   }
 
   generateID() {
+    assert (this._view.generateID);
```
Comment (dstockwell) [inline runtime/viewlet.js]: no space after `assert`

File: runtime/entity.js [hunk 2]
```diff
@@ -11,19 +11,36 @@ const assert = require('assert');
...
+    return this._userIDComponent;
+  }
+
   isIdentified() {
     return this[Symbols.identifier] !== undefined;
   }
   identify(identifier) {
     assert(!this.isIdentified());
     this[Symbols.identifier] = identifier;
+    let components = identifier.split(':');
```
Comment (dstockwell) [inline runtime/entity.js]: this might break when the uid contains `:`

Comment (shans) [reply runtime/entity.js]: I assert the uid component doesn't contain ':'

File: runtime/viewlet.js (hunk 1 above)
Comment (shans) [reply runtime/viewlet.js]: Done.

**Pull request merged**